core/src/components/rating/rating.ts

```diff
@@ -298,7 +320,44 @@ export function createRating(config?: PropsConfig<RatingProps>): RatingWidget {
 				}
 			},
 		},
-		directives: {},
+		directives: {
+			containerDirective: createAttributesDirective(() => ({
+				events: {
+					keydown: (e) => widget.actions.handleKey(e),
+					mouseleave: () => widget.actions.leave(),
+				},
+				attributes: {
+					role: 'slider',
+					class: className$,
+					'aria-valuemin': 0,
+					tabindex: tabindex$,
+					'aria-valuemax': maxRating$,
+					'aria-valuenow': visibleRating$,
+					'aria-valuetext': ariaValueText$,
+					'aria-readonly': computed(() => (readonly$() ? 'true' : undefined)),
+					'aria-disabled': computed(() => (disabled$() ? 'true' : undefined)),
+					'aria-label': ariaLabel$,
+					'aria-labelledby': computed(() => ariaLabelledBy$() || undefined),
+				},
+				classNames: {
+					'au-rating': writable(true),
+				},
+			})),
+			starDirective: createAttributesDirective((starContext$: ReadableSignal<{index: number}>) => {
+				const index = starContext$().index;
+				return {
+					events: {
+						mouseenter: () => widget.actions.hover(index + 1),
+						click: () => widget.actions.click(index + 1),
+					},
+					styles: {
+						cursor: computed(() => (interactive$() ? 'pointer' : 'default')),
+					},
```

Comment:
Should we have the `au-rating-star` class here, as we have `au-rating` in `containerDirective`?